typescript/index.d.ts

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import {
+  FormApi,
+  Config,
+  Decorator,
+  FormState,
+  FormSubscription,
+  FieldSubscription
+} from 'final-form';
+
+export interface ReactContext {
+  reactFinalForm: FormApi;
+}
+
+export interface FieldRenderProps<T extends HTMLElement> {
```

Comment:
Hey @Andarist, can you tell why FieldRenderProps now has the an argument now?

Reply:
Truth to be told - I don't know. I don't recall making this change, but I've looked through git history and my local reflog and it seems I was indeed the one making it.

That being said the change makes sense to me `FieldRenderProps["input"]` is a set of closely related things, the generic should constraint the whole thing and not the specific handlers separately, because you could call them with different input type.

Could you paste in the example code which broke by this change? Could you describe why do you think this change should be reverted? We could now start writing tests to be sure such changes won't affect existing codebases.

However it's somewhat a known problem (and I dont think we can really improve on that as community) for libraries shipping type definitions - personally I dont think that typings change should require a major bump, it would be too restrictive. So the best approach is to lock your dependencies and update frequently.